### PR TITLE
Changes to documentation

### PR DESCRIPTION
**Pull request by Robo210**
Tries to resolve any notes that contain a question.
Fix a few grammatical issues.

### CODE REVIEW

File: documentation/Windows Event Log (EVT) format.asciidoc [hunk 1]
```diff
@@ -168,7 +170,8 @@ The event record (EVENTLOGRECORD) is variable of size and consist of:
 | Offset | Size | Value | Description
 | 0 | 4 | | Size +
 Including the size value
-| 4 | 4 | "LfLe" | Signature (ELF_LOG_SIGNATURE)
+| 4 | 4 | "LfLe" | Signature (ELF_LOG_SIGNATURE) +
+0x654c664c
```
Comment (joachimmetz) [inline documentation/Windows Event Log (EVT) format.asciidoc]: why add 0x654c664c?

File: documentation/Windows Event Log (EVT) format.asciidoc [hunk 2]
```diff
@@ -178,12 +181,16 @@ Contains 32-bit Unix epoch of the date and time in UTC the record was written to
 | 24 | 2 | | Event type
 | 26 | 2 | | Number of strings
 | 28 | 2 | | Event category
-| 30 | 2 | | [yellow-background]*Unknown (Event flags)* +
-Actual usage unknown indicated as reserved
-| 32 | 4 | | [yellow-background]*Unknown (Closing record number)* +
-Actual usage unknown indicated as reserved
+| 30 | 2 | | *ReservedFlags (Event flags)* +
```
Comment (joachimmetz) [inline documentation/Windows Event Log (EVT) format.asciidoc]: Please leave as Unknown, fine that MS-EVEN claims that this is ReservedFlags or event flags, but its purpose is still unknown

File: documentation/Windows Event Log (EVT) format.asciidoc [hunk 3]
```diff
@@ -178,12 +181,16 @@ Contains 32-bit Unix epoch of the date and time in UTC the record was written to
...
+Always 0 +
+`[MS-EVEN]` indicates the value 0x8000 has meaning, but
+no version of the Event Log uses this flag for any purpose
```
Comment (joachimmetz) [inline documentation/Windows Event Log (EVT) format.asciidoc]: "no version of the Event Log uses this flag for any purpose" did you validate this? or is this based on the documentation?

File: documentation/Windows Event Log (EVT) format.asciidoc [hunk 4]
```diff
@@ -178,12 +181,16 @@ Contains 32-bit Unix epoch of the date and time in UTC the record was written to
...
+| 32 | 4 | | *Reserved (Closing record number)* +
+Always 0
```
Comment (joachimmetz) [inline documentation/Windows Event Log (EVT) format.asciidoc]: reword to "MS-EVEN states that this value should always 0"

I've seen multiple times that MS documentations claims one thing but in practice the value is not 0 due to optimization.

**Comment by joachimmetz** [comment]
Thx for the PR, I'll have a more detailed look when time permits. I understand some of your changes but do not agree with all of them. I'll leave some comments.

File: documentation/Windows Event Log (EVT) format.asciidoc [hunk 5]
```diff
@@ -178,12 +181,16 @@ Contains 32-bit Unix epoch of the date and time in UTC the record was written to
...
 | 28 | 2 | | Event category
-| 30 | 2 | | [yellow-background]*Unknown (Event flags)* +
-Actual usage unknown indicated as reserved
-| 32 | 4 | | [yellow-background]*Unknown (Closing record number)* +
-Actual usage unknown indicated as reserved
+| 30 | 2 | | *ReservedFlags (Event flags)* +
+Always 0 +
+`[MS-EVEN]` indicates the value 0x8000 has meaning, but
+no version of the Event Log uses this flag for any purpose
+| 32 | 4 | | *Reserved (Closing record number)* +
```
Comment (joachimmetz) [inline documentation/Windows Event Log (EVT) format.asciidoc]: Leave unknown.

File: documentation/Windows Event Log (EVT) format.asciidoc [hunk 6]
```diff
@@ -210,12 +218,11 @@ This value is used to indicate the end of the event record
 |===
 
 [NOTE]
-The "strings offset" points to the offset of "copy of size" (or "data offset"
-if "data size" is not 0) the strings are empty. [yellow-background]*Can the
-"strings offset" contain 0 ?*
+If the user SID size is 0 then the string offset may equal the user SID offset. +
```
Comment (joachimmetz) [inline documentation/Windows Event Log (EVT) format.asciidoc]: Trailing `+` is not needed

File: documentation/Windows Event Log (EVT) format.asciidoc [hunk 7]
```diff
@@ -291,8 +298,16 @@ path specification of the event message file.
...
-This value can contain multiple filenames separated by a semi colon (;)
-character and that the name of the event message files is case insensitive.
+If set, this value must container one or more filenames separated by a semicolon (;)
+character. The names of the files are case insensitive. Values that are a directory
+or an invalid file (see below) are ignored. Filenames may contain case-insensitive
+environment variables that must be expanded.
+
+[NOTE]
+Each program that displays event message strings is responsible for reading the
+registry value, handling any environment variables within the value, and calling
```
Comment (joachimmetz) [inline documentation/Windows Event Log (EVT) format.asciidoc]: registry => Windows Registry

File: documentation/Windows Event Log (EVT) format.asciidoc [hunk 8]
```diff
@@ -291,8 +298,16 @@ path specification of the event message file.
...
+LoadLibrary on each file. Programs other than the Event Viewer may treat
```
Comment (joachimmetz) [inline documentation/Windows Event Log (EVT) format.asciidoc]: "calling LoadLibrary on each file", this is not a requirement. evtexport does not but can still read the message strings.

File: documentation/Windows Event Log (EVT) format.asciidoc [hunk 9]
```diff
@@ -291,8 +298,16 @@ path specification of the event message file.
...
+environment variables differently.
```
Comment (joachimmetz) [inline documentation/Windows Event Log (EVT) format.asciidoc]: "Programs other than the Event Viewer may treat environment variables differently."

what are you trying to explain with this?

Comment (Robo210) [reply documentation/Windows Event Log (EVT) format.asciidoc]: Below this the doc calls out %SystemRoot% and %WinDir%, and points to where to find these values in the registry. Logging software can use any environment variables it wants in these paths though, and the paths can vary based on the logged-in user. While uncommon, this does mean that tools that try to parse the file format manually or even just using ReadEventLog() may show different messages than Event Viewer. It's a major problem with the whole "each program has to locate and render the messages itself" design of the EVENTLOGRECORD events, and I thought it would be useful to explicitly call that out in case it saves somebody time trying to debug why their messages sometimes don't match Event Viewer's.

File: documentation/Windows Event Log (EVT) format.asciidoc (hunk 8 above)
Comment (Robo210) [reply documentation/Windows Event Log (EVT) format.asciidoc]: Fair point. I can rephrase this.

File: documentation/Windows Event Log (EVT) format.asciidoc (hunk 1 above)
Comment (Robo210) [reply documentation/Windows Event Log (EVT) format.asciidoc]: When looking at the file in a hex editor, it's nice to know the numeric values rather than having to calculate them yourself. Especially if you mess up the little endian bit - the value is ```(DWORD)'eLfL'```, which may not be entirely clear to readers seeing "LfLe".

File: documentation/Windows Event Log (EVT) format.asciidoc (hunk 9 above)
Comment (joachimmetz) [reply documentation/Windows Event Log (EVT) format.asciidoc]: It is even more complicated than that ;)

there is double indirection, see [Parameter expansion](https://github.com/libyal/libevt/blob/master/documentation/Windows%20Event%20Log%20(EVT)%20format.asciidoc#parameter-expansion)

File: documentation/Windows Event Log (EVT) format.asciidoc (hunk 1 above)
Comment (joachimmetz) [reply documentation/Windows Event Log (EVT) format.asciidoc]: Then please add it after the "LfLe" as:

Or `"LfLe" (0x654c664c)`

Then again I've also seen people be confused by hex notation.

**Comment by Robo210** [comment]
I've verified that ReservedFlags and ClosingRecordNumber have never been used by the Event Log going back all the way to NT 3.5.1. They were both added to the structure back in roughly 1993 for a feature that was never implemented.

ClosingRecordNumber is and was always set to 0 when writing to and reading from the file. Unlike ReservedFlags, there is no way that I can find to get a non-0 value without externally editing the file, at which point this document would no longer apply.

ReservedFlags is never used by the service and is set to 0 when logging events through ReportEvent. If external software calls ElfrReportEvent as described in MS-EVEN, it can pass a Flags value that will be copied directly into ReservedFlags in the file, but the value will never be used by the service. The value in the file will be passed on to any callers of ReadEventLog(). So, while this was not the original intent of the field, the effective use of ReservedFlags is to pass 4 bytes of data from the RPC client that logged the event directly to the reader of the event (and the reader would have to know what the intent of the data is - hence MS-EVEN suggesting a particular shared meaning for value 0x8000). A better description for it would therefore be "caller (event writer) defined value".

All that being said, however, I am not aware of any software that logs events to the Event Log through direct RPC calls rather than the Win32 APIs, and so the value will for all intents and purposes always be 0. I can change the description to include as much or as little of this as you want.

Full disclaimer - I work on the Windows Event Log team at Microsoft (but please don't take this as any sort of appeal to authority - I would just like to provide context for how I arrived at these conclusions because you're right that not all of this is clear in the docs, and the docs can't always be trusted). I've simply confirmed that these particular docs match the current and historical behavior in these cases.

**Comment by joachimmetz** [comment]
@Robo210 thanks for the additional context.

> I can change the description to include as much or as little of this as you want.

Maybe add this in a note, from a digital forensic point of view, what is known about specific values can be relevant.

File: documentation/Windows Event Log (EVT) format.asciidoc (hunk 6 above)
Comment (Robo210) [reply documentation/Windows Event Log (EVT) format.asciidoc]: I thought it looked better with the three "if"s on separate lines.

File: documentation/Windows Event Log (EVT) format.asciidoc (hunk 9 above)
Comment (Robo210) [reply documentation/Windows Event Log (EVT) format.asciidoc]: I tried to rephrase this to be more clear that the messages may not match what Event Viewer shows, but I'm not sure if I actually made it better or worse.

Comment (joachimmetz) [reply documentation/Windows Event Log (EVT) format.asciidoc]: Thx, I'll go over the changes (likely later this week) before merge, and make tweaks if necessary.

**Comment by joachimmetz** [comment]
@Robo210 thanks for changes and information. I'll have a look at merging the changes as soon as time permits.

**Review by joachimmetz — APPROVED** [review]
Made some minor additional changes, LGTM
